data/helpers.d/apt

```diff
@@ -266,3 +316,223 @@ ynh_remove_app_dependencies () {
     local dep_app=${app//_/-}	# Replace all '_' by '-'
     ynh_package_autopurge ${dep_app}-ynh-deps	# Remove the fake package and its dependencies if they not still used.
 }
+
+#=================================================
+
+# Install packages from an extra repository properly.
+#
+# usage: ynh_install_extra_app_dependencies --repo="repo" --package="dep1 dep2" [--key=key_url] [--name=name]
+# | arg: -r, --repo - Complete url of the extra repository.
+# | arg: -p, --package - The packages to install from this extra repository
+# | arg: -k, --key - url to get the public key.
+# | arg: -n, --name - Name for the files for this repo, $app as default value.
```

Comment:
For my own culture: can this helper be used also during upgrade ?

E.g. I install mongodb from an extra apt repo ... Then later I want to upgrade the app and its dependency (in the upgrade script). I can and should call this helper again such that it should upgrade the mongodb and that should work, right ?

Reply:
I think so yes, this helper will add your specific repo, upgrade the package, then remove the repo.

Reply:
I confirm that's not the case. I had to specify `"grafana (>=7.0.0)"` to get it to be upgraded from 6.7.3. I think we should come up with something to fix that... but I'm really not comfortable with that piece of code about dependencies... :scream: